src/exoscale/vinyl/fn.clj

```diff
@@ -2,12 +2,38 @@
   "Utility functions to build java.util.function implementations"
   (:import java.util.function.Function
            java.util.function.BiFunction
-           java.util.function.Consumer))
+           java.util.function.Consumer
+           java.util.function.BiConsumer
+           java.util.function.Supplier
+           java.util.concurrent.CompletableFuture
+           java.lang.AutoCloseable))
 
 (defn ^Function make-fun
   "Turn a function of one argument into a java Function"
```

Comment:
The description is not totally accurate here.
To be able to use this function correctly, the parameter `f` needs to have 1-arity AND 2-arities.
Otherwise, you will be able to only use it as a `BiConsumer/BiFunction` or as a `Function/Consumer`.

Reply:
Thanks, I used a small trick with `definterface` to make things a bit cleaner